### PR TITLE
Added a new pattern folder and its related documents “Custom Filter”

### PR DESCRIPTION
**Pull request by sdash-redhat**
The current pull request contains a new pattern design submission "Custom Filter".

### CODE REVIEW

File: pattern-library/forms-and-controls/custom-filter/design/overview.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+## Overview  
+
+Custom Filter helps the user in saving any applied filter criteria with a name so that at a later point of time he can access the same from a list of saved such filter criteria and apply for desired the result set.
+
+
+
+## Use Cases
+
+This pattern can be used where the PatternFly Filter is used to filter a data set on the view. The user while applies different filter criteria in a view, the same can be saved with a name for later usage.
+
```
Comment (catrobson) [inline pattern-library/forms-and-controls/custom-filter/design/overview.md]: I think we want this to really hone in on the "saving" part of the actions, so a suggested reword.  What do you think?

"This pattern should be used where users want to save filters they have built out for future use."

File: pattern-library/forms-and-controls/custom-filter/design/overview.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+
+### Saving a Custom Filter
+![Saving a Filter](img/customfilter-page2.png)
```
Comment (catrobson) [inline pattern-library/forms-and-controls/custom-filter/design/overview.md]: We discovered during implementation of this into patternfly-ng that putting the save action on the right hand side conflicts with some other patterns where we should number of pages and number of results on this area.  It was suggested that we should put the Save on the left after the "Clear Filter" action.  Can you try an update to the design with this?  You can see the problem as discussed by Dan here:  https://github.com/patternfly/patternfly-ng/pull/155

File: pattern-library/forms-and-controls/custom-filter/design/overview.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+This template should provide high level information on the design, to inform the user, at a glance, if this pattern is a fit for their needs.
+
+# Custom Filter
```
[the remark this answers is not in the document]
Comment (sdash-redhat) [reply pattern-library/forms-and-controls/custom-filter/design/overview.md]: Agree. Will change the name and the folder attributes

File: pattern-library/forms-and-controls/custom-filter/design/overview.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+## Overview  
+
+Custom Filter helps the user in saving any applied filter criteria with a name so that at a later point of time he can access the same from a list of saved such filter criteria and apply for desired the result set.
```
[the remark this answers is not in the document]
Comment (sdash-redhat) [reply pattern-library/forms-and-controls/custom-filter/design/overview.md]: This is helpful :)

File: pattern-library/forms-and-controls/custom-filter/design/overview.md (hunk 1 above)
Comment (sdash-redhat) [reply pattern-library/forms-and-controls/custom-filter/design/overview.md]: Makes sense. It goes well with the name and indicates the purpose of the filter clearly!

File: pattern-library/forms-and-controls/custom-filter/design/overview.md (hunk 2 above)
Comment (sdash-redhat) [reply pattern-library/forms-and-controls/custom-filter/design/overview.md]: Reviewed the thread patternfly/patternfly-ng#155. It's clear now. 
Will make the changes.

**Comment by sdash-redhat** [comment]
Updated the files. Renamed the "Custom filter" folder to "Saved Filters" . Updated content, file names and images accordingly along with the changes prescribed. 
Please review. 
Best
Samir

**Comment by dlabrecq** [comment]
FYI, the Travis build failed with the following message.

"Error: This site.md file references a invalid design.md file:
 site.md: pattern-library/forms-and-controls/saved-filter/site.md
 design: pattern-library/{category-name}/{pattern-name}/design/design.md"

**Comment by sdash-redhat** [comment]
Added the section describing the 'delete' interaction as discussed.

**Comment by mcarrano** [comment]
This PR has been inactive for over 6 months and I will close. If this work picks up again in the future, I recommend opening a new PR.

**Pull request closed without merging**